### PR TITLE
Add home basic bottom sheet

### PR DESCRIPTION
**Pull request by raul0208**
Add only the basis structure of the bottom sheet for the Home screen. 

https://user-images.githubusercontent.com/112905642/198136842-9852f171-4049-4c4b-a7a5-c67711568761.mov

### CODE REVIEW

File: ios-target-app/Home/Views/CreateTargetBottomSheetView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    
+    addSubviews([
+      targetImageView,
+      createTargetButton
+    ])
+    
+    layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+    layer.cornerRadius = UI.BottomSheet.cornerRadius
+    layer.shadowRadius = UI.BottomSheet.shadowRadius
+    backgroundColor = .white
```
Comment (kstoletniy) [inline ios-target-app/Home/Views/CreateTargetBottomSheetView.swift:39]: i would encapsulate this in a single func to decouple the code like i.e `configureViews`, also for constrains setup below

File: ios-target-app/Home/Views/CreateTargetBottomSheetView.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    NSLayoutConstraint.activate([
+      targetImageView.topAnchor.constraint(equalTo: self.topAnchor, constant: UI.BottomSheet.defaultTopAnchor),
+      createTargetButton.topAnchor.constraint(equalTo: targetImageView.bottomAnchor)
+    ])
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
```
Comment (kstoletniy) [inline ios-target-app/Home/Views/CreateTargetBottomSheetView.swift:52]: pls, implement this
```suggestion
  required init?(coder: NSCoder) {
    super.init(frame: .zero)
  }
```

**Review by kstoletniy — APPROVED** [review]

File: ios-target-app/Home/Views/CreateTargetBottomSheetView.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  
+  weak var delegate: BottomSheetDelegate?
+  
+  init() {
+    super.init(frame: .zero)
+    
+    addSubviews([
+      targetImageView,
+      createTargetButton
+    ])
```
Comment (glm4) [inline ios-target-app/Home/Views/CreateTargetBottomSheetView.swift:34]:
```suggestion
    addSubviews([targetImageView, createTargetButton])
```

File: ios-target-app/Home/Views/HomeViewController.swift [hunk 4]
```diff
@@ -105,15 +112,32 @@ class HomeViewController: UIViewController, MKMapViewDelegate {
...
+      createTargetBottomSheetView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+      createTargetBottomSheetView.heightAnchor.constraint(equalToConstant: UI.BottomSheet.height)
     ])
   }
 }
+
+extension HomeViewController: BottomSheetDelegate {
+  func createTargetButtonTapped() {
+    let saveTargetBottomSheetViewController = SaveTargetBottomSheetViewController()
+    let navigationController = UINavigationController(rootViewController: saveTargetBottomSheetViewController)
```
Comment (glm4) [inline ios-target-app/Home/Views/HomeViewController.swift:135]: do you really need to embed the view controller into a navigation view controller?

File: ios-target-app/Home/Views/SaveTargetBottomSheetViewController.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    
+    setBottomSheetUIConfigs()
+    configureViews()
+  }
+  
+  func setBottomSheetUIConfigs() {
+    view.layer.cornerRadius = UI.BottomSheet.cornerRadius
+    view.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+    view.layer.shadowColor = UIColor.black.cgColor
+    view.layer.shadowRadius = UI.BottomSheet.shadowRadius
```
Comment (glm4) [inline ios-target-app/Home/Views/SaveTargetBottomSheetViewController.swift:58]: I would create helper functions to add round corners and shadows to any `UIView` so you stay DRY

File: ios-target-app/Home/Views/SaveTargetBottomSheetViewController.swift [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      targetTitleField,
+      topicLabel,
+      topicField,
+      saveButton
+    ])
+    
+    [areaField,
+     targetTitleField,
+     topicField,
+    ].forEach {
```
Comment (glm4) [inline ios-target-app/Home/Views/SaveTargetBottomSheetViewController.swift:79]:
```suggestion
    [areaField, targetTitleField, topicField,].forEach {
```

**Review by glm4 — APPROVED** [review]

**Review by danialepaco — APPROVED** [review]

File: ios-target-app/Home/Views/CreateTargetBottomSheetView.swift (hunk 2 above)
Comment (raul0208) [reply ios-target-app/Home/Views/CreateTargetBottomSheetView.swift:52]: @kstoletniy sure, I think it's already implemented on the line 29, within the init() method.

**Pull request merged**